CRM/Tag/Form/Merge.php

```diff
@@ -81,24 +81,25 @@ public function setDefaultValues() {
    */
   public function postProcess() {
     $params = $this->exportValues();
-    $deleted = CRM_Utils_Array::collect('name', $this->_tags);
+    $names = CRM_Utils_Array::collect('name', $this->_tags);
+    $deleted = array_combine($names, CRM_Utils_Array::collect('label', $this->_tags));
     $primary = array_shift($this->_tags);
 
     foreach ($this->_tags as $tag) {
       CRM_Core_BAO_EntityTag::mergeTags($primary['id'], $tag['id']);
     }
 
-    if ($params['name'] != $primary['name']) {
-      civicrm_api3('Tag', 'create', ['id' => $primary['id'], 'name' => $params['name']]);
+    if ($params['label'] != $primary['label']) {
+      civicrm_api3('Tag', 'create', ['id' => $primary['id'], 'label' => $params['label']]);
     }
 
-    $key = array_search($params['name'], $deleted);
+    $key = array_search($params['name'], $names);
```

Comment:
This doesn't work because `$params['name']` doesn't exist anymore. `$params` is the form's submitted values and the field on the form has been changed and is now called `label`. (And it's hard to see because it's a popup but if you look in the log it says `Warning: Undefined array key "name" in CRM_Tag_Form_Merge->postProcess() (line 96 of .../sites/all/modules/civicrm/CRM/Tag/Form/Merge.php).` so it's definitely not set.)

So there's two options:
1. `$params['label']`
2. Just remove this whole block about unsetting, because it's not really needed.

And then up near line 84 I think all that needed to change was name to label.